src/main/java/tdl/client/queue/QueueBasedImplementationRunner.java

```diff
@@ -103,17 +94,20 @@ public int getRequestTimeoutMillis() {
         //Obtain response from user
         Response response = processingRules.getResponseFor(request);
         audit.log(response);
+        if (response instanceof FatalErrorResponse) {
```

Comment:
This was not here before, so we should not change this behaviour

Reply:
In fact, it is required in some sense, because removing the client actions does cause one specific test to break. The one that checks if the server recovers gracefully in case of an exception.

Its possible the above implementation may need some rework but I have picked it from the previous implementation. In the old implementation when there was an exception raised, a FatalErrorResponse was returned instead of a ValidResponse, how do we intend to handle this now that we have removed the client actions implementation.